solidity/contracts/maintainer/MaintainerProxy.sol

```diff
@@ -0,0 +1,588 @@
+    ///         transaction cost.
+    /// @param redemptionTx Bitcoin redemption transaction data
+    /// @param redemptionProof Bitcoin redemption proof data
+    /// @param mainUtxo Data of the wallet's main UTXO, as currently known on
+    ///        the Ethereum chain
+    /// @param walletPubKeyHash 20-byte public key hash (computed using Bitcoin
+    ///        HASH160 over the compressed ECDSA public key) of the wallet which
+    ///        performed the redemption transaction
+    function submitRedemptionProof(
+        BitcoinTx.Info calldata redemptionTx,
+        BitcoinTx.Proof calldata redemptionProof,
+        BitcoinTx.UTXO calldata mainUtxo,
+        bytes20 walletPubKeyHash
+    ) external onlyMaintainer {
+        uint256 gasStart = gasleft();
+
+        bridge.submitRedemptionProof(
+            redemptionTx,
+            redemptionProof,
+            mainUtxo,
+            walletPubKeyHash
+        );
+
+        reimbursementPool.refund(
+            (gasStart - gasleft()) + _submitRedemptionProofGasOffset,
+            msg.sender
+        );
+    }
+
+    /// @notice Submits the moving funds target wallets commitment.
+    ///         Once all requirements are met, that function registers the
+    ///         target wallets commitment and opens the way for moving funds
+    ///         proof submission.
+    /// @param walletPubKeyHash 20-byte public key hash of the source wallet
+    /// @param walletMainUtxo Data of the source wallet's main UTXO, as
+    ///        currently known on the Ethereum chain
+    /// @param walletMembersIDs Identifiers of the source wallet signing group
+    ///        members
+    /// @param walletMemberIndex Position of the caller in the source wallet
+    ///        signing group members list
+    /// @param targetWallets List of 20-byte public key hashes of the target
+    ///        wallets that the source wallet commits to move the funds to
+    function submitMovingFundsCommitment(
+        bytes20 walletPubKeyHash,
+        BitcoinTx.UTXO calldata walletMainUtxo,
+        uint32[] calldata walletMembersIDs,
+        uint256 walletMemberIndex,
+        bytes20[] calldata targetWallets
+    ) external {
+        uint256 gasStart = gasleft();
+
+        bridge.submitMovingFundsCommitment(
+            walletPubKeyHash,
+            walletMainUtxo,
+            walletMembersIDs,
+            walletMemberIndex,
+            targetWallets
+        );
+
+        reimbursementPool.refund(
+            (gasStart - gasleft()) + _submitMovingFundsCommitmentGasOffset,
+            msg.sender
+        );
+    }
+
+    /// @notice Resets the moving funds timeout for the given wallet if the
+    ///         target wallet commitment cannot be submitted due to a lack
+    ///         of live wallets in the system.
+    /// @param walletPubKeyHash 20-byte public key hash of the moving funds wallet.
+    function resetMovingFundsTimeout(bytes20 walletPubKeyHash) external {
```

Comment:
Shouldn't this one be `onlyMaintainer`? It's a public knowledge transaction...

Reply:
same @pdyraga

Reply:
OK, @pdyraga explained in https://github.com/keep-network/tbtc-v2/pull/209#issuecomment-1139462894 that although it's a public knowledge transaction, it's closely tied to `submitMovingFundsCommitment` and not as expensive as some other transactions - it should be left without `onlyMaintainer` modifier.